src/modules/articles/components/ArticleBody.tsx

```diff
@@ -109,37 +108,71 @@ interface IProps {
   article: IArticle
 }
 
-const ArticleBody: React.FunctionComponent<IProps> = ({ classes, article }) => {
-  // TODO: refactor media to make the carousel part of media/back-end
-  const isCarouselButtonVisible =
-    SPEC_IMG_CAROUSEL_PATTERN.test(article.content) && article.media && article.media.length > 0;
+class ArticleBody extends React.Component<IProps> {
+  constructor(props: IProps) {
+    super(props);
 
-  return (
-    <Row>
-      <Col xs={12} sm={12} md={8} lg={8} className={classes.ArticleBody}>
-        {SPEC_IMG_CAROUSEL_PATTERN.test(article.content) && (
-          <Lightbox title={article.title}>
-            <Gallery media={article.media} />
-          </Lightbox>
-        )}
-        {article.media && article.media.length > 0 && (
-          <ArticleFeaturedMedia
-            image={article.media[0]}
-            isCarouselButtonVisible={isCarouselButtonVisible}
-            carouselImageCount={article.media.length}
-          />
-        )}
-        <ArticleReference article={article} />
-        <div
-          className={classes.innerHTML}
-          dangerouslySetInnerHTML={{ __html: article.content }}
-        />
-      </Col>
-      <Col xsHidden smHidden mdOffset={1} md={3} lgOffset={1} lg={3}>
-        <RightRail />
-      </Col>
-    </Row>
-  );
-};
+    //creates div in which to store article contents
+    this.articleContentDiv = document.createElement("div");
+
+    this.createArticleContent = this.createArticleContent.bind(this);
+  }
+
+  private createArticleContent(element: HTMLDivElement | null): any {
+    //React gives us a DOM element from the render() method if it's been mounted.
+    //We fill this in with article content if it's not null.
+    if (element != null) {
+      element.appendChild(this.articleContentDiv)
+    }
+  }
+
+  private articleContentDiv: HTMLDivElement;
+
+  render() {
+    //Sets the div's contents to hold the HTML provided by the server.
+    this.articleContentDiv.innerHTML = this.props.article.content;
+
+    const { article, classes } = this.props;
+    const isFeaturedGalleryActive =
+      article.media && article.media.length > 0 && SPEC_IMG_CAROUSEL_PATTERN.test(article.content);
+    const featuredMedia = isFeaturedGalleryActive ? article.media : (article.media ? article.media.slice(0, 1) : undefined);
```

Comment:
I'm not in love with this logic. Is there a way to put this logic in the GraphQL server?